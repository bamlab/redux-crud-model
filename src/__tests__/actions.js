import { normalize, schema } from 'normalizr';
import createModuleActionCreators from '../actions';

describe('actions', () => {
  const context = { contextKey: 'contextValue' };
  let actionsCreators;
  beforeEach(() => {
    actionsCreators = createModuleActionCreators({ name: 'myEntity' });
  });

  it('should get the fetchOne action', () => {
    expect(actionsCreators.fetchOne('an-id', context)).toMatchSnapshot();
  });

  it('should get the fetchOneSuccess action', () => {
    const entity = { id: 'an-id', name: 'John Doe' };
    const normalizedEntity = normalize(entity, new schema.Entity('myEntity', {}));
    expect(actionsCreators.fetchOneSuccess(normalizedEntity, context)).toMatchSnapshot();
  });

  it('should get the fetchOneFailure action', () => {
    expect(actionsCreators.fetchOneFailure('an-id', new Error(), context)).toMatchSnapshot();
  });

  it('should get the fetchList action', () => {
    expect(actionsCreators.fetchList('listName', context)).toMatchSnapshot();
  });

  it('should get the fetchListSuccess action', () => {
    const entities = [{ id: 'an-id', name: 'John Doe' }, { id: 'an other id', name: 'Mr Smith' }];
    const mSchema = new schema.Values(new schema.Entity('myEntity', {}));
    const normalizedEntity = normalize(entities, mSchema);
    expect(
      actionsCreators.fetchListSuccess(normalizedEntity, 'listName', context)
    ).toMatchSnapshot();
  });

  it('should get the fetchListFailure action', () => {
    expect(actionsCreators.fetchListFailure(new Error(), 'listName', context)).toMatchSnapshot();
  });
});
