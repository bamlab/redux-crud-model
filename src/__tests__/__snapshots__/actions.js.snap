// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions should get the fetchList action 1`] = `
Object {
  "meta": Object {
    "context": Object {
      "contextKey": "contextValue",
    },
  },
  "payload": Object {
    "name": "listName",
  },
  "type": "MODEL_MYENTITY.FETCH_LIST",
}
`;

exports[`actions should get the fetchListFailure action 1`] = `
Object {
  "error": true,
  "meta": Object {
    "context": Object {
      "contextKey": "contextValue",
    },
    "name": "listName",
  },
  "payload": [Error],
  "type": "MODEL_MYENTITY.FETCH_LIST_FAILURE",
}
`;

exports[`actions should get the fetchListSuccess action 1`] = `
Object {
  "meta": Object {
    "context": Object {
      "contextKey": "contextValue",
    },
    "entities": Object {
      "myEntity": Object {
        "an other id": Object {
          "id": "an other id",
          "name": "Mr Smith",
        },
        "an-id": Object {
          "id": "an-id",
          "name": "John Doe",
        },
      },
    },
  },
  "payload": Object {
    "name": "listName",
    "result": Object {
      "0": "an-id",
      "1": "an other id",
    },
  },
  "type": "MODEL_MYENTITY.FETCH_LIST_SUCCESS",
}
`;

exports[`actions should get the fetchOne action 1`] = `
Object {
  "meta": Object {
    "context": Object {
      "contextKey": "contextValue",
    },
  },
  "payload": Object {
    "id": "an-id",
  },
  "type": "MODEL_MYENTITY.FETCH_ONE",
}
`;

exports[`actions should get the fetchOneFailure action 1`] = `
Object {
  "error": true,
  "meta": Object {
    "context": Object {
      "contextKey": "contextValue",
    },
    "id": "an-id",
  },
  "payload": [Error],
  "type": "MODEL_MYENTITY.FETCH_ONE_FAILURE",
}
`;

exports[`actions should get the fetchOneSuccess action 1`] = `
Object {
  "meta": Object {
    "context": Object {
      "contextKey": "contextValue",
    },
    "entities": Object {
      "myEntity": Object {
        "an-id": Object {
          "id": "an-id",
          "name": "John Doe",
        },
      },
    },
  },
  "payload": Object {
    "result": "an-id",
  },
  "type": "MODEL_MYENTITY.FETCH_ONE_SUCCESS",
}
`;
